File: submissions/dmopc21c1p1.cpp
#include <bits/stdc++.h>

// 4/18/2023
using namespace std;
int n;
int main() {
    cin.sync_with_stdio(0);
    cin.tie(0);
    cin>>n;
    long long alice = 0, duke = 0;
    for (int i = 0; i < n; i++) {
        int a;cin>>a;
        if (a%2==0) {
            duke += ceil(a/2.0);
        } else {
            alice += ceil(a/2.0);
        }
    }
    cout << (alice>=duke?"Alice\n":"Duke\n");
}

File: submissions/dmopc21c1p1.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class PartialGame {
    public static void main(String[] args) throws IOException {
        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in));
        bufferedReader.readLine();
        long duke = 0;
        long alice = 0;
        for (String string : bufferedReader.readLine().split(" ")) {
            long number = Long.parseLong(string);
            if (number % 2 == 0) {
                duke += Math.ceil(number / 2f);
            } else {
                alice += Math.ceil(number / 2f);
            }
        }
        System.out.println(duke > alice ? "Duke" : "Alice");
    }
}